appbuilder/services/npm-service.ts

```diff
@@ -19,6 +21,43 @@ export class NpmService implements INpmService {
 		private $logger: ILogger,
 		private $projectConstants: Project.IConstants) { }
 
+	private get npmBinary(): string {
+		if (!this._npmBinary) {
+			try {
+				require(NpmService.NPM_MODULE_NAME);
+				let npmMainJsFile = require.resolve(NpmService.NPM_MODULE_NAME);
+				let nodeModulesDirName = "node_modules";
+				let pathToNpmBinary = path.join(npmMainJsFile.substring(0, npmMainJsFile.lastIndexOf(nodeModulesDirName) + nodeModulesDirName.length), ".bin", this.npmExecutableName);
+
+				if (!this.$fs.exists(pathToNpmBinary).wait()) {
+					throw new Error(`There npm binary was not found in ${pathToNpmBinary}.`);
+				}
+
+				this._npmBinary = pathToNpmBinary;
+			} catch (err) {
+				if (!this.npmExecutableName) {
```

Comment:
checking if `this.npmExecutableName` returns a truthy result doesn't guarantee that the user has `npm` installed. You'd better make sure by calling the `getNpmVersion` method of the `sysInfoBase` module